File: staffs/src/main/java/com/newfloats/staffs/ui/breaks/ScheduledBreaksFragmnt.java
package com.newfloats.staffs.ui.breaks;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.fragment.app.Fragment;

import com.newfloats.staffs.R;
import com.newfloats.staffs.ui.bottomsheets.BottomSheetFragment;
import com.newfloats.staffs.ui.details.StaffDetailsActivity;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link ScheduledBreaksFragmnt#newInstance} factory method to
 * create an instance of this fragment.
 */
public class ScheduledBreaksFragmnt extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    public static ScheduledBreaksFragmnt newInstance() {
        ScheduledBreaksFragmnt fragment = new ScheduledBreaksFragmnt();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_schedule_breaks, container, false);
        FrameLayout addBreaks = v.findViewById(R.id.fl_add_breaks);
        addBreaks.setOnClickListener(v1 -> {
            BottomSheetFragment bottomSheetFragment = new BottomSheetFragment();
            bottomSheetFragment.show(getActivity().getSupportFragmentManager(), bottomSheetFragment.getTag());
        });
        ((StaffDetailsActivity) requireActivity()).setToolBarTitle("Scheduled Breaks", false);
        return v;
    }

}

File: appService/src/main/java/com/appservice/ui/staffs/ui/breaks/ScheduledBreaksFragmnt.kt
package com.appservice.ui.staffs.ui.breaks

import android.text.Html
import android.view.View
import androidx.fragment.app.Fragment
import com.appservice.R
import com.appservice.base.AppBaseFragment
import com.appservice.databinding.FragmentScheduleBreaksBinding
import com.appservice.ui.staffs.ui.bottomsheets.BottomSheetFragment
import com.framework.models.BaseViewModel

/**
 * A simple [Fragment] subclass.
 * Use the [ScheduledBreaksFragmnt.newInstance] factory method to
 * create an instance of this fragment.
 */
class ScheduledBreaksFragmnt : AppBaseFragment<FragmentScheduleBreaksBinding, BaseViewModel>() {


  companion object {
    private const val ARG_PARAM1 = "param1"
    private const val ARG_PARAM2 = "param2"
    fun newInstance(): ScheduledBreaksFragmnt {
      return ScheduledBreaksFragmnt()
    }
  }

  override fun getLayout(): Int {
    return R.layout.fragment_schedule_breaks
  }

  override fun getViewModelClass(): Class<BaseViewModel> {
    return BaseViewModel::class.java
  }

  override fun onCreateView() {
    super.onCreateView()
    setOnClickListener(binding!!.flAddBreaks)
    binding?.ctvHeading?.text =
      Html.fromHtml(getString(R.string.add_a_leave_break_for_br_gaurav_sharma))
  }

  override fun onClick(v: View) {
    super.onClick(v)
    when (v) {
      binding?.flAddBreaks -> {
        val bottomSheet = BottomSheetFragment()
        bottomSheet.show(requireActivity().supportFragmentManager, null)
      }
    }
  }
}